src/lib/util.ts

```diff
@@ -32,4 +32,45 @@ export default class Util {
         return "Unknown network"
     }
   }
+
+  /**
+   * Performs an Arc.js action.
+   * @param topic Arc.js TransactionService topic to listen to for pending transactions
+   * @param f action to perform
+   * @param opts options to pass to @f
+   * @param onPending callback that's called on every pending transaction
+   * @param onError callback that's called upon any error / failed transaction
+   */
+  public static async performAction<T>(topic: string, f: (opts: any) => Promise<T>, opts: any, onPending: (txCount: number) => any, onError: (e: Error) => any): Promise<T> {
+    let sub: Arc.IEventSubscription;
+    let count = 0;
+    try {
+      const key = Arc.TransactionService.generateInvocationKey(`${topic}.pendingTransactions`);
+      sub = Arc.TransactionService.subscribe(topic, (topic, info) => {
+        if (info.options.key === key && info.tx) {
+          if (Arc.Utils.getWeb3().toDecimal(info.tx.receipt.status) !== 1) {
+            console.error(info.tx);
+            const err = new Error(`Transaction '${info.tx.receipt.transactionHash}' failed on '${topic}' with options '${JSON.stringify(opts, undefined, 2)}'`);
+            console.error(err);
+            onError(err);
+            sub.unsubscribe();
+          } else {
+            onPending(info.txCount);
+            count++;
+            if (count >= info.txCount) {
```

Comment:
How could count be greater than txCount?

Reply:
Can't hurt to check for `>` too :)

Reply:
Thing is, it is confusing, raises questions in the programmers mind as to whether there is a non-obvious case (like it did with me, and I know more about this code than a newbie would).  I think is best in this case to comment as such, or change the code.

Reply:
fixed: https://github.com/daostack/alchemy/pull/61/commits/16f87cd8e057c404cd058d247b3d2cd0bc04f1eb